apps/docs/pages/guides/platform/database-usage.mdx

```diff
@@ -23,20 +23,44 @@ This value is also reported in the [database settings page](https://app.supabase
 ## Database storage management
 
 Supabase uses network-attached storage to balance performance with scalability.
-For Pro and Enterprise projects, disk size expands ~1.5x automatically (e.g., 8GB -> 12GB) when you reach 90% of the disk size.
-Disk size expansion can only occur once every six hours.
-Pro projects can store up to 1024TB.
 
-All projects enter read-only mode when you reach 95% of the disk size. In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
+For Pro and Enterprise tier projects, [disk size expands ~1.5x automatically](#paid-tier-disk-auto-scaling) when you reach 90% of the disk size. Disk size expansion can only occur once every six hours.
+
+For Free tier projects, the database will enter ['read only' mode when you reach 95% disk space](#free-tier-project-read-only-mode).
+
+Pro and enterprise tier projects can store up to 1024TB.
+
+### Free tier project 'read-only' mode
+
+Free tier projects enter read-only mode when you reach 95% of the disk size.
+In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
 Regular operation (read-write mode) is automatically re-enabled once usage is below 95% of the disk size.
 
+For [disk auto-scaling](paid-tier-disk-auto-scaling), and avoid your database entering read-only mode, you can [upgrade to the Pro or Enterprise plan](https://app.supabase.com/project/_/settings/billing/subscription).
+
+### Paid tier disk auto-scaling
+
+Pro and Enterprise have disk auto-scaling. Storage expands automatically when the database reaches 90% of the disk size. The disk is expanded to be 50% larger (e.g., 8GB -> 12GB).
+Auto-scaling can only take place once every 6 hours.
+
+If you intend to import a lot of data into your database that would require multiple storage expansions then [reach out to our team](https://app.supabase.com/support/new), otherwise your project might enter a ['read-only' mode](#paid-tier-project-read-only-mode).
+
 If you need more than 1024TB of disk size or require multiple storage expansions in a short period of time, [contact us](https://app.supabase.com/support/new) to learn more about the Enterprise plan.
```

Comment:
This isn't a great suggestion, but we are talking about 2 different things here, so split it up.

```suggestion
You may require multiple storage expansions in a short period of time. For example, you need to upload more than 1.5x the current size of your database storage. This will lead to your database entering a [read-only mode](#paid-tier-project-read-only-mode). You can avoid this by [contacting the support team](https://app.supabase.com/support/new) so that your database can be resized to a sufficient amount before you start uploading to it.

If you need more than 1024TB of disk size, you can [contact us](https://app.supabase.com/support/new) to learn more about the Enterprise plan. 
```

I think we now also support self serve auto-scaling, or it's coming soon. so need to also update this.

Not that sure we need to keep talking about the 1024TB limit, it always sounds a bit like a negative enforced limit.